packages/core/src/html/linkProcessor.js

```diff
@@ -87,16 +87,41 @@ function convertMdAndMbdExtToHtmlExt(node) {
     if (hasNoConvert) {
       return;
     }
+
     const { href } = node.attribs;
-    const hasMdExtension = href.slice(-3) === '.md';
+
+    const hasMdExtension = href.includes('.md');
     if (hasMdExtension) {
-      const newHref = `${href.substring(0, href.length - 3)}.html`;
+      const mdIdx = href.indexOf('.md');
+      const pathWithoutExtension = href.substring(0, mdIdx);
+
+      const hasURLFragment = href.includes('.md#');
+      if (hasURLFragment) {
+        const fragment = href.substring(mdIdx + 3, href.length);
+        const newHref = `${pathWithoutExtension}.html${fragment}`;
+        node.attribs.href = newHref;
+        return;
+      }
+
+      const newHref = `${pathWithoutExtension}.html`;
       node.attribs.href = newHref;
       return;
     }
-    const hasMbdExtension = href.slice(-4) === '.mbd';
+
+    const hasMbdExtension = href.includes('.mbd');
     if (hasMbdExtension) {
```

Comment:
Similar comment as above - having guard clauses for `hasMbdExtension` and `hasURLFragment`